lib/Driver/Driver.cpp

```diff
@@ -1949,8 +1957,14 @@ void Driver::buildActions(SmallVectorImpl<const Action *> &TopLevelActions,
   }
 
   if (OI.shouldLink() && !AllLinkerInputs.empty()) {
-    auto *LinkAction = C.createAction<LinkJobAction>(AllLinkerInputs,
-                                                     OI.LinkAction);
+    JobAction *LinkAction = nullptr;
+
+    if (OI.LinkAction == LinkKind::StaticLibrary)
+      LinkAction = C.createAction<ArchiveJobAction>(AllLinkerInputs,
+                                                    OI.LinkAction);
+    else
+      LinkAction = C.createAction<LinkJobAction>(AllLinkerInputs,
+                                                 OI.LinkAction);
```

Comment:
Nitpick: I know this is just one *statement* in each branch, but it's now multiple *lines.* Can you add braces?